TESTS/assumptions/SPI/SPI.cpp

```diff
@@ -1,3 +1,8 @@
+// check if SPI is supported on this device
+#if !DEVICE_SPI
+    #error SPI is not supported on this platform, add 'DEVICE_SPI' definition to your platform.
```

Comment:
did you make sure to test your changes on multiple boards to make sure it doesnt break SPI?